app/src/main/java/ch/epfl/skysync/database/tables/AvailabilityTable.kt

```diff
@@ -1,14 +1,29 @@
 package ch.epfl.skysync.database.tables
 
+import ch.epfl.skysync.database.DateLocalDateConverter
 import ch.epfl.skysync.database.FirestoreDatabase
 import ch.epfl.skysync.database.Table
 import ch.epfl.skysync.database.schemas.AvailabilitySchema
 import ch.epfl.skysync.models.calendar.Availability
+import ch.epfl.skysync.models.calendar.AvailabilityStatus
+import ch.epfl.skysync.models.calendar.TimeSlot
+import ch.epfl.skysync.models.flight.Flight
+import ch.epfl.skysync.models.user.User
+import com.google.firebase.firestore.FieldPath
+import com.google.firebase.firestore.Filter
+import java.time.LocalDate
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
 
 /** Represent the "availability" table */
 class AvailabilityTable(db: FirestoreDatabase) :
     Table<Availability, AvailabilitySchema>(db, AvailabilitySchema::class, PATH) {
 
+  private val userTable = UserTable(db)
```

Comment:
Here you have a circular dependency problem, you can not have `AvailabilityTable` in `UserTable` and `UserTable` in `AvailabilityTable`, instead you need to either take `userTable` as argument of the method, or create a third class (for ex: `Queries`) that takes both tables and perfom the queries.

Reply:
I will do a class Queries will all the specific queries.